File: src/types/candle.ts
export interface Candle {
  time: number
  open: number
  high: number
  low: number
  close: number
}

export type TimeFrame = '1' | '5' | '15' | '60' | '240' | 'D' | 'W'

export interface TimeFrameConfig {
  label: string
  resolution: TimeFrame
  intervalMs: number
}

File: crates/core/src/candle.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, specta::Type)]
pub struct Candle {
    #[specta(type = f64)]
    pub time: i64,
    pub open: f64,
    pub high: f64,
    pub low: f64,
    pub close: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize, specta::Type)]
#[serde(rename_all = "camelCase")]
pub struct CandlesResponse {
    pub candles: Vec<Candle>,
    pub regular_market_price: f64,
    pub previous_close: f64,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, specta::Type)]
#[serde(rename_all = "lowercase")]
pub enum CandleSource {
    Db,
    Yahoo,
}
